common/predictive-text/worker/correction/context-tracker.ts

```diff
@@ -351,8 +351,11 @@ namespace correction {
       }
 
       const hasDistribution = transformDistribution && Array.isArray(transformDistribution);
-      const primaryInput = hasDistribution ? transformDistribution[0].sample : null;
-      const isBackspace = primaryInput && primaryInput.insert == "" && primaryInput.deleteLeft > 0;
+      let primaryInput = hasDistribution ? transformDistribution[0].sample : null;
+      if(primaryInput && primaryInput.insert == "" && primaryInput.deleteLeft == 0 && !primaryInput.deleteRight) {
+        primaryInput = null;
+      }
+      const isBackspace = primaryInput && primaryInput.insert == "" && primaryInput.deleteLeft > 0 && !primaryInput.deleteRight;
```

Comment:
The different comparisons between `deleteLeft` and `deleteRight` here make me feel a bit squirmy. Is there a reason for them (and if so, probably should comment the difference).

Reply:
Sure.

`0 == undefined` returns `false`.

Reply:
Compare against this:

https://github.com/keymanapp/keyman/blob/58a2bda5344a4fa153f772614d644f30b919ec46/common/models/types/index.d.ts#L226-L256

`deleteLeft` must be defined, but not `deleteRight`.